src/core/server/elasticsearch/elasticsearch_service.ts

```diff
@@ -76,7 +78,7 @@ export class ElasticsearchService
 
     const config = await this.config$.pipe(first()).toPromise();
 
-    this.getAuthHeaders = deps.http.getAuthHeaders;
+    this.authHeaders = deps.http.authRequestHeaders;
     this.executionContextClient = deps.executionContext;
     this.client = this.createClusterClient('data', config);
```

Comment:
__issue:__ I'm implementing Security part on top of this PR in https://github.com/elastic/kibana/pull/122155 and noticed that this client is created _before_ `unauthorizedHandler` is configured. If I'm not mistaken, that means that all non-custom clients (e.g. the one provided via HTTP handler context) won't be supported?

Reply:
Yea, you're right, atm this base client doesn't have the handler registered. This needs to be addressed

Reply:
Should be handled by https://github.com/elastic/kibana/pull/120677/commits/f6370f413172821ea8e06dfc0d80ff5b5a6e90bd, if you mind taking another look